docs(data table): usage update

designertyler opened the pull request:
Do not merge. For review only

This is the data table usage docs update.

Preview link https://carbon-website-git-fork-designertyler-docsdatatablemay12.carbon-design-system.now.sh/components/data-table/usage

aagonzales commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+| Type             | Purpose                                                                                                                                                                                                                             |
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| _Basic_          | The basic data table comes with a base style with only the title, header, and table elements. rows.                                                                                                                                 |
+| _Zebra_          | The zebra data table features alternating colored rows to make scanning horizontal information easier for the user.                                                                                                                 |
```
Move zebra stripes to "modifier" section since its a style change and not a UX feature.

aagonzales commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+### Sizing
 
-### Search
+The data table is available in 4 different row sizes: compact, short, normal, tall.
```
Link to the style tab here where there is more information. See the modal page so similar scenario. 
/components/data-table/style#row-sizes

aagonzales commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+<Column colLg={8}>
 
-### Expandable
+![data table row hover](images/datatable-hover.png)
```
This example looks a little like a zebra strip and not a hover. I think maybe add a 4th row to help delineate the hover row from the normal rows.

aagonzales commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+### Pagination
 
-The expandable data table is useful for presenting large amounts of data in a small space. Use the expanded section for supplementary information or data that needs additional query time.
+Pagination divides table data into separate pages. Pagination is accompanied by an option that enables the user to change the number of items per page. The pagination component is always placed at the bottom of the data table. See the [pagination](/components/pagination/code) component for further guidelines and configurations.
```
"Advanced* pagination is accompanied by the option ... " You show two options but you don't talk about the two options.

aagonzales commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+<Row>
+<Column colLg={12}>
+
+![data table multi select](images/datatable-multi-select.png)
```
This image is showing expandable table not batch action.

aagonzales commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+</Column>
+</Row>
+
+## Content
```
In the new updates to the template the content section should go after the "formatting" section and before "behaviors"

designertyler replied:
I'll update to the latest template format. This change pushes the behaviors section further down on the usage tab. Wouldn't the content section be better on the style tab?

laurenmrice commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
-## Additional functionality
+| Type             | Purpose                                                                                                                                                                                                                             |
+| ---------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| _Basic_          | The basic data table comes with a base style with only the title, header, and table elements. rows.                                                                                                                                 |
```
This is referred to as "Default" instead of Basic in code.

laurenmrice commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+- As a replacement for a spreadsheet application.
 
-<Caption>Basic data table</Caption>
+### Variations
```
“Types”  should be the header instead of Variations.

laurenmrice commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
+5. **Pagination table bar**: an optional component that lets a user navigate data as pages when the amount of data it too large to be shown at once.
 
-<Caption>Data table with various Pagination methods</Caption>
+### Sizing
```
The the different sizes you mention in text should be in order of how you are showing it in the images.

laurenmrice commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -277,148 +280,312 @@ import {
-<div className="image--fixed">
+The wide grid mode is the default and gives the most breathing room between the data table and the other components or content.
 
-![Table toolbar](images/data-table-usage-6.png)
```
It would be good to mention somewhere (maybe the sizing section?) that the toolbar and the batch actions bar come in two different sizes depending on which height of data table you are using. Jan might have recommendations on where to put this content.

laurenmrice commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -12,8 +12,11 @@ Data tables are used to organize and display data efficiently. The data table co
 
-<AnchorLink>Basic data table</AnchorLink>
-<AnchorLink>Additional functionality</AnchorLink>
+<AnchorLink>Overview</AnchorLink>
```
"Live Demo" should be an anchor link under Overview

laurenmrice commented on src/pages/components/data-table/usage.mdx:
```diff
@@ -12,8 +12,11 @@ Data tables are used to organize and display data efficiently. The data table co
+<AnchorLink>Formatting</AnchorLink>
+<AnchorLink>Behavior</AnchorLink>
+<AnchorLink>Content</AnchorLink>
+<AnchorLink>Related</AnchorLink>
```
It would be good if there was a "References" section of validated articles under Related.

janchild commented:
@designertyler I will do a more detailed read-through but as a first step, could you please place the content into the format of the component template. The latest templates are here: https://www.carbondesignsystem.com/how-to-contribute/component#documentation-templates

You can also see the template in action in these components: 
- https://www.carbondesignsystem.com/components/checkbox/usage
- https://www.carbondesignsystem.com/components/modal/usage
- https://www.carbondesignsystem.com/components/notification/usage

Thanks!

jeanservaas commented:
@designertyler could you update these images? @laurenmrice and I changed the way we're using stokes to denote crops. We're using a 1px Gray 20 stroke now (instead of .5px G30). It's kind of a fussy change but makes a difference, so I just quickly updated the images that it would affect and attached them here. I'm also attaching the new sketch file. Once these are updated (oh! and I see the animation) I will approve. Not sure what else is pending for content. 

[data_table_usage_js.sketch 2.zip](https://github.com/carbon-design-system/carbon-website/files/4814414/data_table_usage_js.sketch.2.zip)


[Data_table_new_imgs.zip](https://github.com/carbon-design-system/carbon-website/files/4814392/Data_table_new_imgs.zip)

andrea-gm commented:
Hello!
I was wondering if it would be good to include use cases for when a designer/developer might want to use the overflowMenuOnHover prop?

Also, does that prop affect the action icons as well or only the overflow icon?

<img width="785" alt="Screen Shot 2020-06-25 at 1 46 37 PM" src="https://user-images.githubusercontent.com/14081333/85794177-070bb900-b6eb-11ea-91eb-194299eb68ab.png">

andrea-gm commented:
Hi again!
Will the react component support a way to turn off the hover state of the rows? I don't see a prop for that currently. It would be good to call it out here in the documentation, where keeping the hover state is mentioned as a consideration.

<img width="774" alt="Screen Shot 2020-06-25 at 1 54 19 PM" src="https://user-images.githubusercontent.com/14081333/85794497-9749fe00-b6eb-11ea-8354-e70a5d583321.png">

andrea-gm commented:
And lastly!
Would you consider adding a note about using links in the data table, specifically when they are used as a way to "drill down" into the asset?

I am happy to provide the content for that if you like the idea!! It could be something like [this section](https://pages.github.ibm.com/cdai-design/pal/wip/data-tables/usage/#linking-from-data-tables) in the CD&AI guideline.

<img width="365" alt="Screen Shot 2020-06-25 at 1 58 49 PM" src="https://user-images.githubusercontent.com/14081333/85794976-5dc5c280-b6ec-11ea-82b0-ab400c02d4c2.png">

andrea-gm commented:
Hi 😬 
The image for **Multi-select / batch action** doesn't show checkboxes. It looks like a duplicate of the image for **Expandable**.

<img width="1183" alt="Screen Shot 2020-06-25 at 4 30 12 PM" src="https://user-images.githubusercontent.com/14081333/85806251-5cec5b00-b703-11ea-808d-b9198c9a5bdb.png">

jeanservaas commented:
Hey @designertyler 

Sorry for the delay, Just looks like two of the Gray 20 strokes are missing

Can you just add a 1px, Gray 20 stroke where the highlights here to this image:

<img width="1243" alt="image" src="https://user-images.githubusercontent.com/43144260/86180086-0f9c2f00-baf1-11ea-89ed-50fdc44db103.png">

And update the sizing image near the top with this image:

![datatable_sizes_new](https://user-images.githubusercontent.com/43144260/86180225-50944380-baf1-11ea-9f03-ba034c886ec1.png)

THEN I AM READY TO APPROVE!!!!!

designertyler commented:
Thanks for the feedback @andrea-gm ! I updated the docs to clarify the overflow prop use case and the row hover language and fixed the duplicate image.

The link behavior you shared didn't mention drill down, but that sounds like something to add under the content section of this page. Did you have more around that or any other feedback?

andrea-gm commented:
Thanks for addressing all that @designertyler ! 

Couple of follow ups. First, regarding the `overflowMenuOnHover` do we know if it affects the action icons as well as the overflow menu icon? Seems like we would need that functionality for the action icons if the goal is to avoid visual clutter. @joshblack, maybe you know more about the implementation of the prop?

And second, you're right! The link I shared doesn't really talk about "drill down". So maybe we need two separate things, (1) a mention of [Linking from data tables](https://pages.github.ibm.com/cdai-design/pal/wip/data-tables/usage/#linking-from-data-tables) and (2) specific guidelines for how to "drill down" into an asset. A guideline for (2) would help settle some of the confusion around [this issue](https://github.ibm.com/CloudIntegrationDesign/DesignSystemAdoptionGuild/issues/105). Maybe there's a way to combine (1) and (2)? Would love to hear your thoughts on this!

jeanservaas commented:
@designertyler thanks for making the updates. This is TINY, but could you nudge the G20 boarder rule on the bottom of the right side of this image up so it's all in the frame, it looks like part of it is getting cut off. (And actually just check both on the bottom to make sure they're all the way in the frame.

I'm approving anyway. Thanks so much!

<img width="1056" alt="image" src="https://user-images.githubusercontent.com/43144260/86372230-85b2aa00-bc47-11ea-9a20-ef47a5b6458b.png">
